src/purchase-selection/utils.ts

```diff
@@ -0,0 +1,155 @@
+  PurchaseSelectionType,
+} from './types';
+import {TariffZoneWithMetadata} from '@atb/tariff-zones-selector';
+import turfBooleanPointInPolygon from '@turf/boolean-point-in-polygon';
+import {UserProfileWithCount} from '@atb/fare-contracts';
+import {TariffZone, UserProfile} from '@atb-as/config-specs';
+import {isValidDateString} from '@atb/utils/date';
+
+export const isProductSellableInApp = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+) => {
+  if (
+    (product.limitations.appVersionMin &&
+      compareVersion(product.limitations.appVersionMin, input.appVersion) >
+        0) ||
+    (product.limitations.appVersionMax &&
+      compareVersion(product.limitations.appVersionMax, input.appVersion) < 0)
+  )
+    return false;
+
+  if (
+    product.distributionChannel.some((channel) => channel === 'debug-app') &&
+    input.customerProfile?.debug
+  )
+    return true;
+
+  return product.distributionChannel.some((channel) => channel === 'app');
+};
+
+export const getDefaultProduct = (
+  input: PurchaseSelectionBuilderInput,
+  configType: string,
+) =>
+  input.preassignedFareProducts
+    .filter((product) => isProductSellableInApp(input, product))
+    .filter((product) => product.type === configType)
+    .reduce((selected, current) => (current.isDefault ? current : selected));
+
+export const getDefaultZone = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): TariffZoneWithMetadata => {
+  const selectableZones = input.tariffZones.filter((zone) =>
+    isSelectableZone(product, zone),
+  );
+
+  if (input.currentCoordinates) {
+    const {longitude, latitude} = input.currentCoordinates;
+    const zoneFromLocation = selectableZones.find((t) =>
+      turfBooleanPointInPolygon([longitude, latitude], t.geometry),
+    );
+    if (zoneFromLocation) {
+      return {...zoneFromLocation, resultType: 'geolocation'};
+    }
+  }
+
+  const zone = selectableZones.reduce((selected, current) =>
+    current.isDefault ? current : selected,
+  );
+  return {...zone, resultType: 'zone'};
+};
+
+export const getDefaultUserProfiles = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): UserProfileWithCount[] => {
+  const selectableProfiles = getSelectableUserProfiles(
+    input.userProfiles,
+    product,
+  );
+  const profile = selectableProfiles.reduce((selected, current) =>
+    current.userTypeString === input.defaultUserTypeString ? current : selected,
+  );
+  return [{...profile, count: 1}];
+};
+
+export const getSelectableUserProfiles = (
+  userProfiles: UserProfile[],
+  product: PreassignedFareProduct,
+): UserProfile[] =>
+  userProfiles.filter((profile) => isSelectableProfile(product, profile));
+
+export const isSelectableProduct = (
+  input: PurchaseSelectionBuilderInput,
+  selection: PurchaseSelectionType,
+  product: PreassignedFareProduct,
+) => {
+  if (selection.fareProductTypeConfig.type !== product.type) return false;
```

Comment:
It might be clear just by reading the code, but only if you know the context. the parameter selection is preselected in the UI? you do not want to include himself as selectable even it can be selectable? I think this util should only respond to isProductSellableInApp, the decision of selection must be done higher, and this encapsulates a logic that does not belong here.

Also would be worth it to add a small documentation in the method itself to know what selection and input are.

Reply:
> It might be clear just by reading the code, but only if you know the context. the parameter selection is preselected in the UI?

It is the current purchase selection, which should be what is presented to the user. I renamed the parameter to `currentSelection` to make it more clear.

> I think this util should only respond to isProductSellableInApp, the decision of selection types is not the same must be done higher level as a first validation, and this encapsulates a logic that does not belong here.

You are right that there needs to be some logic regarding this higher in the app. This module doesn't cover presentation logic, and what product options to display for a given type config is logic that also needs to be present in the presentation layer. However, it must still be checked here as well, since one of the features of this new module is that it guarantees the validity of the purchase selection it creates, so it can't rely on checks being done outside of the module.

> Also would be worth it to add a small documentation in the method itself to know what selection and input are.

As mentioned I renamed the `selection` parameter to `currentSelection`. Regarding `input`, that parameter is used throughout the whole module. Instead of documenting the param, I added some doc on the `PurchaseSelectionBuilderInput` type itself.

Reply:
I see, good to know and for the details, at first glance it gives different impressions.